app/models/service_interaction.rb

```diff
@@ -1,4 +1,12 @@
 class ServiceInteraction < ActiveRecord::Base
   validates :service_identifier, :service_label, :interaction_identifier,
   :interaction_label, presence: true, uniqueness: true
+
+  def self.find_service(identifier)
+    Service.find_by identifier: identifier
+  end
+
+  def self.find_interaction(identifier)
+    Interaction.find_by identifier: identifier
+  end
```

Comment:
@h-lame Would there be any drawbacks to using `belongs_to` / `has_one` for this now?